File: android/app/src/main/java/app/tourism/ui/common/buttons/PrimaryButton.kt
package app.tourism.ui.common.buttons

import ButtonLoading
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import app.tourism.ui.theme.TextStyles

@Composable
fun PrimaryButton(
    modifier: Modifier = Modifier,
    label: String,
    onClick: () -> Unit,
    isLoading: Boolean = false,
    enabled: Boolean = true,
    backgroundColor: Color = MaterialTheme.colorScheme.primary
) {
    Button(
        modifier = Modifier.fillMaxWidth().height(56.dp).then(modifier),
        onClick = onClick,
        enabled = enabled,
        shape = RoundedCornerShape(16.dp),
        colors = ButtonDefaults.buttonColors(containerColor = backgroundColor),
        elevation = ButtonDefaults.buttonElevation(defaultElevation = 0.dp)
    ) {
        Box(modifier = Modifier.padding(vertical = 3.dp)) {
            if (isLoading)
                ButtonLoading()
            else
                ButtonText(buttonLabel = label)
        }
    }
}

@Composable
fun ButtonText(buttonLabel: String) {
    Text(
        text = buttonLabel,
        style = TextStyles.h4,
        fontWeight = FontWeight.W700,
        maxLines = 1,
        overflow = TextOverflow.Ellipsis,
        color = MaterialTheme.colorScheme.onPrimary
    )
}

File: iphone/Maps/Tourism/Presentation/Components/Buttons/PrimaryButton.swift
import SwiftUI

struct PrimaryButton: View {
  var label: String
  var onClick: () -> Void
  var isLoading: Bool = false
  var enabled: Bool = true
  
  var body: some View {
    Button(action: onClick) {
      HStack {
        if isLoading {
          ProgressView()
            .progressViewStyle(CircularProgressViewStyle())
        } else {
          Text(label)
            .font(.headline)
            .fontWeight(.bold)
            .foregroundColor(Color.onPrimary)
            .padding()
            .frame(maxWidth: .infinity)
        }
      }
      .frame(
        maxWidth: .infinity,
        minHeight: 50
      )
      .background(Color.primary)
      .cornerRadius(16)
    }
    .disabled(!enabled || isLoading)
  }
}
